packages/wallet-sdk/src/connection/RxWebSocket.ts

```diff
@@ -132,7 +110,7 @@ export class RxWebSocket<T = object> {
     const { webSocket } = this;
     if (!webSocket) {
       this.pendingData.push(data);
-      this.connect().subscribe();
+      this.connect().then();
```

Comment:
do we need the `.then()` can we just do `void this.connect();`

Reply:
this was to actually execute the promise that `this.connect()` returns, as `subscribe()` did.
i think one alternative could be `await this.connect()`, but didn't go with that since it would change the logic. this whole function used to just initiate the connection without waiting for it to complete and return.

Reply:
actually you were right. i think i got confused while switching between rxjs and promises. 😭  updated.